src/main/java/com/google/cloud/tools/managedcloudsdk/ManagedCloudSdk.java

```diff
@@ -176,15 +181,49 @@ public SdkUpdater newUpdater() {
   }
 
   /** Get a new {@link ManagedCloudSdk} instance for @{link Version} specified. */
-  public static ManagedCloudSdk newManagedSdk(Version version) throws UnsupportedOsException {
+  public static ManagedCloudSdk newManagedSdk(Version version)
+      throws UnsupportedOsException, InvalidOsStateException {
+    OsInfo osInfo = OsInfo.getSystemOsInfo();
     return new ManagedCloudSdk(
         version,
-        Paths.get(System.getProperty("user.home"), ".google-cloud-tools-java", "managed-cloud-sdk"),
-        OsInfo.getSystemOsInfo());
+        getOsSpecificManagedSdkHome(osInfo.name(), System.getProperties(), System.getenv()),
+        osInfo);
   }
 
   /** Convenience method to obtain a new LATEST {@link ManagedCloudSdk} instance. */
-  public static ManagedCloudSdk newManagedSdk() throws UnsupportedOsException {
+  public static ManagedCloudSdk newManagedSdk()
+      throws UnsupportedOsException, InvalidOsStateException {
     return newManagedSdk(Version.LATEST);
   }
+
+  @VisibleForTesting
+  static Path getOsSpecificManagedSdkHome(
+      OsInfo.Name osName, Properties systemProperties, Map<String, String> environment)
+      throws InvalidOsStateException {
+    Path userHome = Paths.get(systemProperties.getProperty("user.home"));
+    Path cloudSdkPartialPath = Paths.get("google-cloud-tools-java", "managed-cloud-sdk");
+
+    switch (osName) {
+      case WINDOWS:
+        String localAppData = environment.get("LOCALAPPDATA");
+        if (localAppData == null || localAppData.trim().isEmpty()) {
+          throw new InvalidOsStateException("LOCALAPPDATA environment is invalid or missing");
```

Comment:
could we instead fall back to the directory we were using before?

Reply:
I agree:
  - I just tested on MacOS and creating a new user populates a host of new directories including `~/Library/Application Support` even before they ever login.  It's extremely unlikely for `~/Library/Application Support` to not exist.
  - `LOCALAPPDATA` is documented in MSDN as being present.  It's extremely unlikely to not exist.

So I think logging an error and falling back to the XDG-style `~/.cache/google-cloud-tools-java` should be great.  And I think we can then get rid of the `InvalidOsStateException`, I think.

Reply:
+1, in favor of getting rid of another checked exception.

Reply:
went to .cache fallback, and removed exception.